jwst/resample/resample.py

```diff
@@ -274,59 +277,69 @@ def resample_many_to_many(self, input_models):
-            # Use this for defining the output filename
-            indx = exposure[0].meta.filename.rfind('.')
-            output_type = exposure[0].meta.filename[indx:]
-            output_root = '_'.join(exposure[0].meta.filename.replace(
-                output_type, '').split('_')[:-1])
-            if self.asn_id is not None:
-                output_model.meta.filename = f'{output_root}_{self.asn_id}_outlier_i2d{output_type}'
-            else:
-                output_model.meta.filename = f'{output_root}_outlier_i2d{output_type}'
 
-            # Initialize the output with the wcs
-            driz = gwcs_drizzle.GWCSDrizzle(output_model, pixfrac=self.pixfrac,
-                                            kernel=self.kernel, fillval=self.fillval)
+            copy_asn_info_from_library(input_models, output_model)
 
-            log.info(f"{len(exposure)} exposures to drizzle together")
-            for img in exposure:
-                img = datamodels.open(img)
-                iscale = self._get_intensity_scale(img)
-                log.debug(f'Using intensity scale iscale={iscale}')
+            with input_models:
+                example_image = input_models.borrow(indices[0])
 
-                inwht = resample_utils.build_driz_weight(
-                    img,
-                    weight_type=self.weight_type,
-                    good_bits=self.good_bits
-                )
-
-                # apply sky subtraction
-                blevel = img.meta.background.level
-                if not img.meta.background.subtracted and blevel is not None:
-                    data = img.data - blevel
+                # Determine output file type from input exposure filenames
+                # Use this for defining the output filename
+                indx = example_image.meta.filename.rfind('.')
+                output_type = example_image.meta.filename[indx:]
+                output_root = '_'.join(example_image.meta.filename.replace(
+                    output_type, '').split('_')[:-1])
+                if self.asn_id is not None:
+                    output_model.meta.filename = f'{output_root}_{self.asn_id}_outlier_i2d{output_type}'
                 else:
-                    data = img.data
+                    output_model.meta.filename = f'{output_root}_outlier_i2d{output_type}'
+                input_models.shelve(example_image, indices[0], modify=False)
+                del example_image
+
+                # Initialize the output with the wcs
+                driz = gwcs_drizzle.GWCSDrizzle(output_model, pixfrac=self.pixfrac,
+                                                kernel=self.kernel, fillval=self.fillval)
+
+                log.info(f"{len(indices)} exposures to drizzle together")
+                for index in indices:
+                    img = input_models.borrow(index)
+                    img = datamodels.open(img) # must call this explicitly to get area reference file
```

Comment:
What's the area reference file?

Reply:
When `datamodels.open()` is not called explicitly here, a bunch of regtest errors that look [like this one](https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1645/testReport/junit/jwst.regtest/test_nirspec_mos_spec3/_stable_deps__test_nirspec_mos_spec3_b000000030_crf_/) happen.  This is due to the fact that the AREA extension (among other data and metadata) does not get added to the datamodel.  I assumed the reason was a failure to access a reference file but I could certainly be wrong about that.  I don't know exactly what's going on here (shouldn't borrowing the model call `datamodels.open()`?) but this way worked so I just left it.  Let me know if you have any alternative ideas.  Note that this line also exists on master.

Reply:
Something weird is happening here. I assume the `datamodels.open` on main is because `ModelContainer` can sometimes provide filenames (when this step is run with `in_memory=False`).

My initial hunch is that it was hiding a bug. The regtest truth file contains 4x `AREA` HDUs (that look to be all 0's from the diff). I will try to run it locally to see what's going. Do you know if the `photom` step is run as part of this regtest? I believe that's the step that should be adding an `AREA` HDU.

Reply:
I think the presence of these `AREA` HDUs in the truth file is a bug on main. They are the result of this line in the `SlitModel` constructor:
https://github.com/spacetelescope/stdatamodels/blob/ee4756ffe9ff71c1fb7b067f8998949a88d6a703/src/stdatamodels/jwst/datamodels/slit.py#L154
When `datamodels.open` is called with a `SlitModel` as an argument, it will call `SlitModel.__init__` providing the open `SlitModel` as an argument. Inside the `__init__` if the original `SlitModel` doesn't have an `area` attribute one will be created due to the line above.
Is there a photom step/spec3 expert who can chime in to determine if this file should contain `AREA` hdus?

Reply:
@jemorrison @melanieclarke are the resident spectroscopy experts on team Chartreuse

Reply:
The Slit models I have worked with include MIRI fixed_slit and NIRSpec FIXEDSLIT and MSA. I may not understand the question full. But the photom step uses a pixel area reference file and photom reference file. NIRSpec and MIRI use this data a bit differently.  I can see why the SlitModel would need an area  defined in the schema.
Not sure if that is the question.

Reply:
Thanks!

@emolter I agree that preventing this PR from changing the regtests is a good goal and deferring addressing this issue to another PR makes sense.

Would you test is the following allows the regtests to pass (this worked locally for me):
```diff
-                    img = datamodels.open(img)
+                    if isinstance(img, datamodels.SlitModel):
+                        img.area = img.area
```
It avoids the clone and hopefully reproduces the bug.

Reply:
I opened a JIRA ticket [here](https://jira.stsci.edu/browse/JP-3709). @jemorrison hopefully this clarifies a bit.. I was wrong, it doesn't have anything to do with the reference files.